hazelcast-client/src/main/java/com/hazelcast/client/proxy/NearCachedClientMapProxy.java

```diff
@@ -554,11 +554,13 @@ public void clear() {
     }
 
     @Override
-    protected void onDestroy() {
-        removeNearCacheInvalidationListener();
-        getContext().getNearCacheManager().destroyNearCache(name);
-
-        super.onDestroy();
+    protected void postDestroy() {
+        try {
+            removeNearCacheInvalidationListener();
```

Comment:
I just saw the different orders in client and member proxies. You aligned those already, so everything is fine for me.